### PR TITLE
Support the "null" version ID

### LINKED ISSUE #241: Allow `S3Path` handle version ID `"null"`
When S3 versioning is enabled in a bucket, objects that were created in that bucket before versioning was enabled have a version ID of `"null"`. Currently, `S3Path` does not support version ID's that are `"null"` even though this is a valid ID.  For example, doing something like this is valid:

```
>> aws s3api get-object --bucket my-bucket --key my/key.txt --version-id null local/path/key.txt
```

From the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html):

> Objects that are stored in your bucket before you set the versioning state have a version ID of null. When you enable versioning, existing objects in your bucket do not change. What changes is how Amazon S3 handles the objects in future requests.

### PR DESCRIPTION
**Pull request by omus**
Fixes #241

### CODE REVIEW

File: test/s3path.jl [hunk 1]
```diff
@@ -554,6 +557,52 @@ function s3path_tests(config)
             @test !exists(versioned_path)
             @test length(s3_list_versions(config, bucket_name, key)) == 1
         end
+
+        @testset "S3Path null version" begin
+            b = let df = dateformat"yyyymmdd\THHMMSS\Z"
+                "ocaws.jl.test." * lowercase(Dates.format(now(Dates.UTC), df))
+            end
```
Comment (omus) [inline test/s3path.jl:564]: Would be nice to use a different named bucket for this. Not sure what restrictions are available so we'll start here

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Review by mattBrzezinski — APPROVED** [review]
LGTM, I just need to step out but can do another pass when I return.

**Comment by omus** [comment]
bors try

File: src/s3path.jl [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+# Amazon states that version IDs are UTF-8 encoded, URL-ready, opaque strings no longer than
+# 1024 bytes
+# – https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html#version-ids
+#
+# In practise version IDs seems to be much narrower in scope:
+# https://github.com/JuliaCloud/AWSS3.jl/pull/199#issuecomment-901995960
+#
+# An unversioned object can be accessed using the "null" version ID. For details see:
+# https://github.com/JuliaCloud/AWSS3.jl/issues/241
+const VERSION_ID_REGEX = r"^(?:[0-9a-zA-Z\._]{32}|null)$"
```
[the remark this answers is not in the document]
Comment (omus) [reply src/s3path.jl:10]: For complicated regular expressions I think having a dedicated test set is the right approach: https://github.com/JuliaLang/julia/blob/7b395153e80672f8cdb18f51dd653a85e28b2070/stdlib/LibGit2/test/libgit2.jl#L237-L328.

In this particular case I think we're not quite complex enough to really require it but it never hurts.

**Pull request merged**